### PR TITLE
Speed

### PR DESCRIPTION
**Pull request by wbunting**

### CODE REVIEW

File: model/model.wl [hunk 1]
```diff
@@ -99,24 +99,28 @@ BetaMeanSig[mu_,sig_]:=BetaDistribution[(mu^2-mu^3-mu sig)/sig,((-1+mu) (-mu+mu^
...
+RandomVariate[PosNormal[importlength0,1]],
+RandomVariate[PosNormal[initialInfectionImpulse0,3]],
+cutoff,
+stateParams["params"]["pS"],
+stateParams["params"]["pH"],
+stateParams["params"]["pC"],
+containmentThresholdRatio0,
+stateParams["icuCapacity"],
+stateParams["hospitalCapacity"]
+}&/@Range[numberOfSimulations]]
```
Comment (khenisey) [inline model/model.wl:123]: Why the switch to an array away from an association?

File: model/model.wl [hunk 2]
```diff
@@ -371,11 +357,18 @@ ParametricNDSolveValue[{
 	{r0natural, importtime}
 ];
 
+endTime[ifun_]:=Part[ifun["Domain"],1,-1];
+
 
 
+setDistancing[state_,scenario_] := Module[{},
+  distancing[t_] := stateDistancing[state, scenario["id"], t];
+];
```
Comment (khenisey) [inline model/model.wl:366]: This seems odd to me. Why call this function instead of just writing that line out in the code?

File: model/model.wl [hunk 3]
```diff
@@ -389,17 +382,104 @@ evaluateScenario[state_, fitParams_, sims_, stateParams_, scenario_]:=Module[{
...
-	distance[t_] := stateDistancing[state, scenario, t];
-	
-	(* do one solution with the mean param values for the estimate *)
-	{sol,evts}=CovidModel[
-	fitParams["r0natural"],
+  
+    Clear[distancing, pfunODE2];
+    distancing[t_] := stateDistancing[state, scenario["id"], t];
+    Clear[Sq,Eq,ISq,RSq,IHq,HHq,RHq,RepHq,Iq,ICq,EHq,HCq,CCq,RCq,Deaq,PCR,est];
+Clear[r0natural,daysUntilNotInfectiousOrHospitalized,daysFromInfectedToInfectious,daysToLeaveHosptialNonCritical,pPCRNH,pPCRH,daysTogoToCriticalCare,daysFromCriticalToRecoveredOrDeceased,fractionOfCriticalDeceased,importtime,importlength,initialInfectionImpulse,tmax,pS,pH,pC,containmentThresholdCases,icuCapacity,hospitalCapacity];
```
Comment (khenisey) [inline model/model.wl:397]: Why clearing all these variables? Could we instead defining them as module scope in the module definition?

File: model/model.wl [hunk 4]
```diff
@@ -426,64 +508,64 @@ evaluateScenario[state_, fitParams_, sims_, stateParams_, scenario_]:=Module[{
...
-	    stateParams["params"]["pC"],
-        containmentThresholdRatio0,
-        stateParams["icuCapacity"],
-        distance,
-        stateParams["hospitalCapacity"]
-    ]&/@sims;
-    
-    deciles = {1/10,2/10,3/10,4/10,5/10,6/10,7/10,8/10,9/10};
+	(* Quiet because we handle bad interpolating functions in the next line *)
+	rawSimResults=Apply[pfunODE2, #]&/@sims//Quiet;
```
Comment (khenisey) [inline model/model.wl:515]: Ah, I think this answers my question about the array v association from above.

File: model/model.wl (hunk 1 above)
Comment (wbunting) [reply model/model.wl:123]: This had to do with how the PDE wants arguments to be applied (at least how rob had written it). I think there is some way I could get the values out of the association inline, but decided not to mess with it for now

File: model/model.wl (hunk 2 above)
Comment (wbunting) [reply model/model.wl:366]: This is probably extraneous now -- will remove

File: model/model.wl (hunk 3 above)
Comment (wbunting) [reply model/model.wl:397]: I was getting weird issues where the definition of the parametric ode wouldn't respect things like the distancing function changing without defining these in the module and clearing. Maybe they could just be module scope though

File: model/model.wl (hunk 4 above)
Comment (wbunting) [reply model/model.wl:515]: Yea...

**Pull request merged**